src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -208,6 +208,32 @@ public DocumentsOperationContext GetContextForNewDocument()
             public void Dispose()
             {
                 FinishBatchOfDocuments();
+                DeleteRevisionsForNonExistingDocuments();
+            }
+
+            private async void DeleteRevisionsForNonExistingDocuments()
+            {
+                if (_missingDocumentsForRevisions == null)
+                    return;
+                using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    foreach (var docId in _missingDocumentsForRevisions)
+                    {
+                        using (DocumentIdWorker.GetSliceFromId(context, docId.Key, out var lowerId))
+                        {
+                            var changeVector = _database.DocumentsStorage.GetNewChangeVector(context, _database.DocumentsStorage.GenerateNextEtag());
+                            var command = new RevisionsOperations.DeleteRevisionCommand(docId.Key,
+                                lowerId,
+                                docId.Value,
+                                changeVector,
+                                DateTime.Now.Ticks,
```

Comment:
need to be _database.Time.GetUtcNow()... Also probably can be handled inside the command.